File: api/routers/system.py
from fastapi import APIRouter, Query
from pydantic import BaseModel

class HealthResponse(BaseModel):
    status: str
    version: str
    data_count: int

router = APIRouter(prefix="/system", tags=["system"])

@router.get("/health", response_model=HealthResponse)
def health_check():
    from api.dependencies import load_jobs_df
    df = load_jobs_df()
    return {"status": "healthy", "version": "1.0.0", "data_count": len(df)}

File: web/src/stores/system.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import api from '@/api'

export const useSystemStore = defineStore('system', () => {
  const healthy = ref(false)
  const dataCount = ref(0)
  const version = ref('')

  async function checkHealth() {
    try {
      const { data } = await api.get('/system/health')
      healthy.value = data.status === 'healthy'
      dataCount.value = data.data_count
      version.value = data.version
    } catch {
      healthy.value = false
    }
  }

  return { healthy, dataCount, version, checkHealth }
})
